### PR TITLE
Design picker: Display a screenshot instead of mshots when externally managed

### LINKED ISSUE #3447: People: Do not display UI for editing WP.com site owner
Currently, when an admin attempts to edit the site owner of a site, an error notice is displayed. Since we don't allow editing, we shouldn't even show the form that would allow editing a site owner.

Originally reported in #3140. I fixed the issue in the API in r131094-wpcom.

### PR DESCRIPTION
**Pull request by WBerredo**
Resolves https://github.com/Automattic/dotcom-forge/issues/3447
Related to D118900-code

## Proposed Changes

* Check if the theme for a given design is externally managed and display a screenshot instead of mshots
* Remove the filter out for Marketplace producs
* Query the Product List to receive Marketplace themes prices

## Testing Instructions

* Apply the diff D118900-code to your sandbox and direct requests to the sandbox
* Go to the route `/setup/update-design/designSetup?siteSlug=:site`
* The Marketplace Themes (ex: Yuna, Olsen) should have the partner tag and properly show the screenshot
* Verify the price is available when hovering the partners tag

CleanShot 2023-08-14 at 14 13 19@2x

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by gcsecsey — APPROVED** [review]
Changes LGTM and work great:
CleanShot 2023-08-15 at 14 23 57@2x

**Comment by WBerredo** [comment]
On hold due to:

https://github.com/Automattic/dotcom-forge/issues/3447#issuecomment-1679276195

**Comment by WBerredo** [comment]
@gcsecsey can you give a new round of review with the updated test instructions?

**Comment by WBerredo** [comment]
This PR is not blocked anymore because it was decoupled from https://github.com/Automattic/wp-calypso/pull/80446 and can be shipped separately as soon as its reviewed.

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (gcsecsey) [inline packages/design-picker/src/components/unified-design-picker.tsx:41]: Nit: Where do these numbers come from? Will these need to be updated in the future?

Currently the image fits nicely for all breakpoints:
| Breakpoint | Page |
|--------|--------|
| <320px | CleanShot 2023-08-21 at 11 45 20@2x |
| <768px | CleanShot 2023-08-21 at 11 45 37@2x |
| <1024px | CleanShot 2023-08-21 at 11 46 04@2x |

**Review by gcsecsey — APPROVED** [review]
Changes LGTM, test well and work as described.

Hovering/touching the partner tag works great on both mobile and desktop:
| Viewport | Screenshot |
|--------|--------|
| mobile | CleanShot 2023-08-21 at 11 45 06@2x |
| desktop | CleanShot 2023-08-21 at 11 42 05@2x |

**Review by mmtr — COMMENTED** [review]
> Remove the following code from the line 320: tier: '-marketplace',

I think it's safe to include that as part of this PR. That part of the logic does not control the themes that are visible in the grid, but the information that it's needed to correctly render the badges and tooltips.

That way, I think we can merge PRs/diffs in this order:

1. https://github.com/Automattic/wp-calypso/pull/80573 (this PR, plus the removal of `tier: '-marketplace'`).
  1.1. Nothing will happen: Partner themes won't be visible since the `starter-designs` endpoint doesn't return then yet
  1.2. Because of the above, we can abandon https://github.com/Automattic/wp-calypso/pull/80446
2. https://github.com/Automattic/wp-calypso/pull/80837
3. More additional PRs
4. D118900-code
  4.1. This is where Partner themes will become visible.

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (WBerredo) [reply packages/design-picker/src/components/unified-design-picker.tsx:41]: Good point, I brought it from here where we also display the screenshot: https://github.com/Automattic/wp-calypso/blob/trunk/client/components/theme/index.jsx#L176

I didn't invest much in figuring out a way to share this value because it comes from `client/components`, and this is a separate package so it must have all the info by itself or be passed by param. In this case, I decided just to replicate the code.

**Comment by WBerredo** [comment]
@mmtr Thanks for the suggestion, and it's a possibility I considered but decided against it, in the end, it would be nice to have other input.

**Pro**

* Its more straightforward 
* It will help us in future reviews by removing the necessity of changing this line

**Con**

* This will query unnecessary data (Marketplace themes) for now, even if we are not displaying it.

---

I'm good with both options, but I would like to put here my line of thought.

**Review by mmtr — APPROVED** [review]
> This will query unnecessary data (Marketplace themes) for now, even if we are not displaying it.

That's fair! Note that we can also add a new feature flag. That way we don't fetch partner themes data in production by default, but can be enabled in development, so we don't need to remove that line.

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (gcsecsey) [reply packages/design-picker/src/components/unified-design-picker.tsx:41]: Thanks for the context. It's good to know that the thumbnails are also a fixed aspect ratio. I wondered what was the original motivation behind moving the design picker to a separate package as it seems to cause some friction when we make updates. I've only found this point about the i18n wrapper: https://github.com/Automattic/wp-calypso/pull/51226#discussion_r597764782

**Comment by WBerredo** [comment]
> Note that we can also add a new feature flag. 

Aha! Good point, I'll add it.

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (WBerredo) [reply packages/design-picker/src/components/unified-design-picker.tsx:41]: >  I wondered what was the original motivation behind moving the design picker to a separate package

I already asked me the same thing, but I didn't find an answer yet.

**Comment by WBerredo** [comment]
The feature flag logic for querying themes is added to this PR after the merge of https://github.com/Automattic/wp-calypso/pull/80446 and the rebase of this PR.

**Pull request merged**
